### PR TITLE
enhance: remote slack attachment

### PR DESCRIPTION
**Pull request by claire1618**
These changes allows to Slack could source a message file from another repo.

This will help is someone has to send a message held in one repo across multiple repos.

Below is an example for the new parameters introduced (remote, registry):
```
parameters:
    filepath: $remote-filepath
    remote: true
    registry: https://github.com
```

### CODE REVIEW

**Review by wass3rw3rk — CHANGES_REQUESTED** [review]
we can probably fix the couple of linter errors in main.go. we got rid of io/ioutil elsewhere for example, we can use the correct module in main.

File: cmd/vela-slack/plugin.go [hunk 1]
```diff
@@ -233,3 +248,63 @@ func getAttachmentFromFile(p *Plugin) ([]slack.Attachment, error) {
...
+	bStr = strings.Replace(bStr, "{{ .BuildFinished }}", strconv.Itoa(p.Env.BuildFinished), -1)
+	bStr = strings.Replace(bStr, "{{ .BuildNumber }}", strconv.Itoa(p.Env.BuildNumber), -1)
+	bStr = strings.Replace(bStr, "{{ .BuildParent }}", strconv.Itoa(p.Env.BuildParent), -1)
+	bStr = strings.Replace(bStr, "{{ .BuildStarted }}", strconv.Itoa(p.Env.BuildStarted), -1)
+	bStr = strings.Replace(bStr, "{{ .RepositoryTimeout }}", strconv.Itoa(p.Env.RepositoryTimeout), -1)
+	bytes = []byte(bStr)
+
+	logrus.Info("pip")
+	logrus.Info(bStr)
+	logrus.Info("pip")
```
Comment (wass3rw3rk) [inline cmd/vela-slack/plugin.go:299]: are these leftover from debugging?

**Comment by ecrupper** [comment]
Can we get these updates to `DOCS.md` since we pull those in for our doc site? Maybe in a separate PR so it isn't updated prematurely?

File: cmd/vela-slack/plugin.go [hunk 2]
```diff
@@ -189,6 +204,24 @@ func (p *Plugin) Validate() error {
 	return nil
 }
 
+// getRemoteAttachment function to open and parse slack attachment json file into
+// slack webhook message payload.
+func replaceString(bytes []byte, p *Plugin) []byte {
+	// Converts bytes into string and replaces {{ .BuildCreated }}
+	// with a timestamp before returning it back into bytes again.
```
Comment (wass3rw3rk) [inline cmd/vela-slack/plugin.go:211]:
```suggestion
// replaceString converts bytes into string and replaces variables
// like {{ .BuildCreated }} with their respective values before
// returning them back into bytes again.
func replaceString(bytes []byte, p *Plugin) []byte {
```

the comment preceding the function didn't seem accurate. the comment within seemed appropriate though. what do you think?

Comment (claire1618) [reply cmd/vela-slack/plugin.go]: oops I forgot to change that, fixed now

**Review by wass3rw3rk — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]
neat

**Pull request merged**